config/plotdefs/dht-availability-classified-overall.yaml

```diff
@@ -3,25 +3,35 @@ datasets:
   - name: "main"
     source: "nebula_ipfs"
     query: >
-      with u as (
-        with period as (
-          select d::date-'1 week'::interval AS start, d::date as end, tstzrange(d::date-'1 week'::interval, d::date) as range, extract(epoch from '1 week'::interval) as duration
-          from generate_series({{ .StartOfWeek | timestamptz }}-'15 week'::interval, {{ .StartOfWeek | timestamptz }}, '1 week'::interval) d
-        ) -- 10 weeks
+      with period as (-- 10 weeks
+        select
+          d::date-'1 week'::interval AS start,
+          d::date as end, tstzrange(d::date-'1 week'::interval, d::date) as range,
+          extract(epoch from '1 week'::interval) as duration
+        from generate_series({{ .StartOfWeek | timestamptz }}-'15 week'::interval, {{ .StartOfWeek | timestamptz }}, '1 week'::interval) d
+      ), u as (
         select period.end as date, s.peer_id,
           extract(epoch from sum(upper(s.uptime * period.range) - lower(s.uptime * period.range ))::interval)/period.duration as uptime
         from period, sessions s
         where s.uptime && period.range
         group by period.end, s.peer_id, period.duration
+      ), offline as (
+        select
+          period.end as date,
+          v.peer_id
+        from period, visits v
+        where v.visit_started_at <@ period.range
+        group by period.end, v.peer_id
+        having bool_and(v.connect_error IS not null)
       )
-      select date,
+      select u.date,
         sum(case when uptime >= 0.8 then 1 else 0 end) as online,
         sum(case when uptime >= 0.4 and uptime < 0.8 then 1 else 0 end) as mostlyonline,
-        sum(case when uptime >= 0.1 and uptime < 0.4 then 1 else 0 end) as mostlyoffline,
-        sum(case when uptime < 0.1 then 1 else 0 end) as offline
-      from u
-      group by date
-      order by date;
+        sum(case when uptime < 0.4 then 1 else 0 end) as mostlyoffline,
```

Comment:
note, redefined mostlyoffline. If we're okay with that we need to update the copy as well

Reply:
I don't think we should redefine.

Reply:
so then we must add to the count 👍

Reply:
why not union with something like

```
select 0 as online, 0 as mostlyonline, 0 as mostlyoffline, count(distinct o.peer_id) as offline
from select
          period.end as date,
          v.peer_id
        from period, visits v
        where v.visit_started_at <@ period.range
        group by period.end, v.peer_id
        having bool_and(v.connect_error IS not null)
```

II haven't tried so possibly we'd have to repeat the `with period`  part too